src/scheduler/queue_info.c

```diff
@@ -326,7 +326,7 @@ query_queues(status *policy, int pbs_sd, server_info *sinfo)
 	pbs_statfree(queues);
 	free_schd_error(sch_err);
 	if (err) {
-		if (qinfo_arr != NULL)
+		if (qinfo_arr[0] != NULL)
```

Comment:
You should leave this one alone.  If qinfo_arr[0] == NULL, then qinfo_arr has memory allocated to it.  This was correct before.

Reply:
I think the check is redundant then. At line 323 we are already dereferencing qinfo_arr so Ideally, we don't even need a check before freeing it up.

Reply:
All the checks are redundant.  I didn't know it at the time I wrote the code, but free(NULL) is a noop.  You don't need these.  Until we go through and remove them all, I'd rather leave them in for consistency.